ostd/src/arch/x86/mm/mod.rs

```diff
@@ -156,14 +158,15 @@ macro_rules! parse_flags {
 
 impl PageTableEntryTrait for PageTableEntry {
     fn is_present(&self) -> bool {
-        self.0 & PageTableFlags::PRESENT.bits() != 0
+        // For PT child, `PRESENT` should be set.
+        // For huge page, `PAT` should be set.
+        // We don't prevent the protecting an absent entry to present.
+        self.0 & PageTableFlags::PRESENT.bits() != 0 || self.0 & PageTableFlags::HUGE.bits() != 0
```

Comment:
I am little bit comfused here. In the docs of HUGE bit it says "We use this bit it in all levels to indicate that this entry is 'valid'", and the logics here means that there may be a case that the PRESENT bit is true and HUGE bit is false. So in this case, the pt is present but not valid. I am confused about what the valid bit represents here.

Reply:
LOL, the documentation and the implementation mismatches. No wonder if you are confused. Sorry.